cni/network/network_test.go

```diff
@@ -1103,3 +1103,75 @@ func TestGetPodSubnetNatInfo(t *testing.T) {
 		require.Empty(t, natInfo, "linux podsubnet natInfo should be empty")
 	}
 }
+
+func TestPluginSwiftV2Add(t *testing.T) {
+	plugin, _ := cni.NewPlugin("name", "0.3.0")
+
+	localNwCfg := cni.NetworkConfig{
+		CNIVersion:                 "0.3.0",
+		Name:                       "swiftv2",
+		ExecutionMode:              string(util.V4Overlay),
+		EnableExactMatchForPodName: true,
+		Master:                     "eth0",
+	}
+
+	tests := []struct {
+		name       string
+		plugin     *NetPlugin
+		args       *cniSkel.CmdArgs
+		wantErr    bool
+		wantErrMsg string
+	}{
+		{
+			name: "SwiftV2 Add Happy path",
+			plugin: &NetPlugin{
+				Plugin:      plugin,
+				nm:          acnnetwork.NewMockNetworkmanager(),
+				ipamInvoker: NewMockIpamInvoker(false, false, false, true, false),
+				report:      &telemetry.CNIReport{},
+				tb:          &telemetry.TelemetryBuffer{},
+			},
+			args: &cniSkel.CmdArgs{
+				StdinData:   localNwCfg.Serialize(),
+				ContainerID: "test-container",
+				Netns:       "test-container",
+				Args:        fmt.Sprintf("K8S_POD_NAME=%v;K8S_POD_NAMESPACE=%v", "test-pod", "test-pod-ns"),
+				IfName:      eth0IfName,
+			},
+			wantErr: false,
+		},
+		{
+			name: "SwiftV2 Add fail",
+			plugin: &NetPlugin{
+				Plugin:      plugin,
+				nm:          acnnetwork.NewMockNetworkmanager(),
+				ipamInvoker: NewMockIpamInvoker(false, false, false, true, true),
+				report:      &telemetry.CNIReport{},
+				tb:          &telemetry.TelemetryBuffer{},
+			},
+			args: &cniSkel.CmdArgs{
+				StdinData:   localNwCfg.Serialize(),
+				ContainerID: "test-container",
+				Netns:       "test-container",
+				Args:        fmt.Sprintf("K8S_POD_NAME=%v;K8S_POD_NAMESPACE=%v", "test-pod", "test-pod-ns"),
+				IfName:      eth0IfName,
+			},
+			wantErr: true,
+		},
+	}
+
```

Comment:
I like to have testcase where defaultNic succeeded but secondary failed and cni should tear down dfaultNic setup and cleanup all allocated resources and vice-versa(secondary succeeded but default failed)

Reply:
added via mock endpoint client, will add via transparent and secondary endpoint client in the future

Reply:
can you create work item for it? This test case is must for cni release